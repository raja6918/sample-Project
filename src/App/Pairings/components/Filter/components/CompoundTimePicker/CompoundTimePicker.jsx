import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Grid } from '@material-ui/core';
import DateFnsUtils from '@date-io/date-fns';
import { TimePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import moment from 'moment';

const PickersWrapper = styled.div`
  .picker-field {
    margin-top: 5px;
    & div > input {
      cursor: pointer;
    }
  }
  .picker-field2 {
    margin-top: 16px;
    & div > input {
      cursor: pointer;
    }
  }
`;

class CompoundTimePicker extends Component {
  state = {
    startTime: this.props.value.startTime
      ? moment(this.props.value.startTime, this.props.displayTimeFormat)
      : null,
    endTime: this.props.value.endTime
      ? moment(this.props.value.endTime, this.props.displayTimeFormat)
      : null,
  };

  componentDidMount() {
    if (this.props.value.noChange) {
      this.sendTime();
    }
  }

  sendTime = () => {
    const { startTime, endTime } = this.state;
    const { returnTimeFormat, onChange } = this.props;
    const startTimeF = moment(startTime).format(returnTimeFormat);
    const endTimeF = moment(endTime).format(returnTimeFormat);
    const value = {};

    if (startTime) value.startTime = startTimeF;
    if (endTime) value.endTime = endTimeF;

    onChange(value, null, null);
  };

  handleChange = (event, field) => {
    this.setState({ [field]: event }, this.sendTime);
  };

  render() {
    const { t, displayTimeFormat, containerStyle } = this.props;
    const { startTime, endTime } = this.state;
    return (
      <Grid container style={containerStyle}>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <PickersWrapper>
            <Grid item xs={12} sm={12}>
              <TimePicker
                name="startTime"
                ampm={false}
                className="picker-field"
                label={t('FILTER.pane.labels.between')}
                format={displayTimeFormat}
                placeholder={displayTimeFormat.toUpperCase()}
                value={startTime}
                onChange={event => this.handleChange(event, 'startTime')}
                InputProps={{
                  endAdornment: (
                    <AccessTimeIcon
                      color="primary"
                      style={{ cursor: 'pointer' }}
                    />
                  ),
                }}
              />
            </Grid>
            <Grid item xs={12} sm={12}>
              <TimePicker
                className="picker-field2"
                name="endTime"
                ampm={false}
                label={t('FILTER.pane.labels.and')}
                format={displayTimeFormat}
                placeholder={displayTimeFormat.toUpperCase()}
                value={endTime}
                onChange={event => this.handleChange(event, 'endTime')}
                InputProps={{
                  endAdornment: (
                    <AccessTimeIcon
                      color="primary"
                      style={{ cursor: 'pointer' }}
                    />
                  ),
                }}
              />
            </Grid>
          </PickersWrapper>
        </MuiPickersUtilsProvider>
      </Grid>
    );
  }
}

CompoundTimePicker.propTypes = {
  t: PropTypes.func.isRequired,
  value: PropTypes.shape({
    startTime: PropTypes.oneOfType([PropTypes.shape({}), null]),
    endTime: PropTypes.oneOfType([PropTypes.shape({}), null]),
    noChange: PropTypes.bool,
  }),
  onChange: PropTypes.func.isRequired,
  returnTimeFormat: PropTypes.string,
  displayTimeFormat: PropTypes.string,
  containerStyle: PropTypes.shape({}),
};

CompoundTimePicker.defaultProps = {
  returnTimeFormat: 'HH:mm',
  displayTimeFormat: 'HH:mm',
  value: {
    startTime: null,
    endTime: null,
    noChange: true,
  },
  containerStyle: { width: '160px' },
};

export default CompoundTimePicker;
