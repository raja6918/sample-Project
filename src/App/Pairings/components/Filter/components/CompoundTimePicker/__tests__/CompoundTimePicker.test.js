import React from 'react';
import { shallow } from 'enzyme';
import CompoundTimePicker from '../CompoundTimePicker';

jest.mock('moment', () => {
  return () => jest.requireActual('moment')('2020-01-01T00:00:00.000Z');
});

describe('Test CompoundTimePicker Component', () => {
  const minProps = {
    t: jest.fn(),
    onChange: jest.fn(),
    returnTimeFormat: 'HH:mm',
    displayTimeFormat: 'HH:mm',
    value: {
      startTime: '00:00',
      endTime: '23:59',
    },
    containerStyle: { width: '160px' },
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<CompoundTimePicker {...minProps} />);
  });

  test('CompoundTimePicker Component renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test('isTimeRangeValid returns true by when time are not set', () => {
  //   wrapper.instance().setState({
  //     startTime: null,
  //     endTime: null,
  //   });
  //   const result = wrapper.instance().isTimeRangeValid();
  //   expect(result).toBeTruthy();
  // });

  // test('isTimeRangeValid returns false when the start date is equal to the end date', () => {
  //   wrapper.instance().setState({
  //     startTime: '01:00',
  //     endTime: '01:00',
  //   });
  //   const result = wrapper.instance().isTimeRangeValid();
  //   expect(result).toBeFalsy();
  // });

  // test('isTimeRangeValid returns true when the start time is lesser than the end time', () => {
  //   wrapper.instance().setState({
  //     startTime: '05:00',
  //     endTime: '07:00',
  //   });
  //   const result = wrapper.instance().isTimeRangeValid();
  //   expect(result).toBeTruthy();
  // });

  test('handleChange unsets the error when the date values are valid', () => {
    wrapper.instance().setState({
      startTime: '02:00',
      endTime: '01:00',
    });
    const event = '05:00';
    const field = 'endTime';
    wrapper.instance().handleChange(event, field);
    expect(wrapper.state('error')).toBeFalsy();
  });
});
