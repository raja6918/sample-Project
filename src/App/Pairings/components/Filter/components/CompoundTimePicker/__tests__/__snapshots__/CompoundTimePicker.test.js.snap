// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CompoundTimePicker Component CompoundTimePicker Component renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  style={
    Object {
      "width": "160px",
    }
  }
>
  <MuiPickersUtilsProvider
    utils={[Function]}
  >
    <CompoundTimePicker__PickersWrapper>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={12}
        xs={12}
      >
        <PickerWithState
          InputProps={
            Object {
              "endAdornment": <Memo
                color="primary"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              />,
            }
          }
          ampm={false}
          className="picker-field"
          format="HH:mm"
          invalidDateMessage="Invalid Time Format"
          name="startTime"
          onChange={[Function]}
          openTo="hours"
          placeholder="HH:MM"
          value={"2020-01-01T00:00:00.000Z"}
          views={
            Array [
              "hours",
              "minutes",
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={12}
        xs={12}
      >
        <PickerWithState
          InputProps={
            Object {
              "endAdornment": <Memo
                color="primary"
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              />,
            }
          }
          ampm={false}
          className="picker-field2"
          format="HH:mm"
          invalidDateMessage="Invalid Time Format"
          name="endTime"
          onChange={[Function]}
          openTo="hours"
          placeholder="HH:MM"
          value={"2020-01-01T00:00:00.000Z"}
          views={
            Array [
              "hours",
              "minutes",
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </CompoundTimePicker__PickersWrapper>
  </MuiPickersUtilsProvider>
</WithStyles(ForwardRef(Grid))>
`;
